shared/types/config/port_modes.go

```diff
@@ -0,0 +1,55 @@
+package config
+
+import "fmt"
+
+// Enum to describe the mode for the RPC port.
+// Closed will not allow any connections to the RPC port.
+// OpenLocalhost will allow connections from the same host.
+// OpenExternal will allow connections from external hosts.
+const (
+	RPC_Closed        RPCMode = "closed"
+	RPC_OpenLocalhost RPCMode = "localhost"
+	RPC_OpenExternal  RPCMode = "external"
+)
+
+func (rpcMode RPCMode) String() string {
+	return string(rpcMode)
+}
+
+func (rpcMode RPCMode) Open() bool {
+	return rpcMode == RPC_OpenLocalhost || rpcMode == RPC_OpenExternal
+}
+
+func (rpcMode RPCMode) DockerPortMapping(port uint16) string {
+	ports := fmt.Sprintf("%d:%d/tcp", port, port)
+
+	if rpcMode == RPC_OpenExternal {
+		return ports
+	}
+
+	if rpcMode == RPC_OpenLocalhost {
+		return fmt.Sprintf("127.0.0.1:%s", ports)
+	}
+
+	return ""
+}
+
+func PortModes(warningOverride string) []ParameterOption {
```

Comment:
Maybe not something that would pass joe's sniff test, but if you make this variadic then the default path looks cleaner.

ie
```golang
func PortModes(warningOverride ...string) []ParameterOption {
    var warning string
    if len(warningOverride) == 0 {
        warning = "Allow connections from external hosts. This is safe if you're running your node on your local network. If you're a VPS user, this would expose your node to the internet"
    } else {
        warning = warningOverride[0]
    }
```

then, elsewhere, instead of
`conf.PortModes("")` you just call `conf.PortModes()` unless there is an override, in which case you pass it as before.

Reply:
I did a little cleanup to make it more palatable but didn't make it variadic cause comparing against "" has parity with other places in the code... wen refactor?